jupyter_ascending/handlers/jupyter_notebook.py

```diff
@@ -215,6 +215,14 @@ def _transform_jupytext_cells(jupytext_cells) -> List[Dict[str, Any]]:
 
     comm.send({"command": "start_sync_notebook", "cells": _transform_jupytext_cells(result["cells"])})
 
+    # Wait for the merge_complete flag to get set in the callback.
+    # This way we don't release the lock before syncing is done.
+    for i in range(500):
+        if merge_complete:
+            return
+        time.sleep(0.01)
+    else:
+        J_LOGGER.warning("Timed out waiting for syncing to complete.")
```

Comment:
This feels weird. Why would this happen? What should the user do about it?

Reply:
Idk, but the alternative is hanging the kernel forever, which seems worse? It's like, I wouldn't expect to see it, but I'd rather handle it cleanly and log it than hang without any indication of what went wrong, or crash the kernel over a (probably recoverable) syncing failure.

Reply:
That's one possible alternative, which I agree does seem worse.

However, another alternative (which I just came up with) would be to notify the client that this happened? Seems much better to have this happen on the client side and create an error message (and non-zero exit code) there, since then the user will actually see it?

Reply:
We could do that, but currently there's no mechanism to report errors back from the kernel to the server (and mostly the kernel isn't even able to know if errors happened - before this PR it was all "fire and forget" type logic, and still mostly is except for this new codepath). Since this error message doesn't seem super critical to handle carefully, this feels to me like another "come back and improve if this gets popular" type of thing - lmk if you concur.

Reply:
perfect, please make an issue for it!

Reply:
created